sdk/ai/Azure.AI.Inference/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -0,0 +1,247 @@
+            GenAiClientOperationDurationMetricName, "s", "Measures GenAI operation duration.");
+        private static readonly Histogram<long> s_tokens = s_meter.CreateHistogram<long>(
+            GenAiClientTokenUsageMetricName, "{token}", "Measures the number of input and output token used.");
+
+        private Activity m_activity;
+        private Stopwatch m_duration;
+        private TagList m_commonTags;
+        private StreamingRecordedResponse m_recordedStreamingResponse;
+
+        /// <summary>
+        /// Return the content of the message.
+        /// Content accessor is implemented only on the exact class, but not on the abstract generated class.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        public static string GetContent(ChatRequestMessage message)
+        {
+            if (message.GetType() == typeof(ChatRequestAssistantMessage))
+                return ((ChatRequestAssistantMessage)message).Content;
+            if (message.GetType() == typeof(ChatRequestSystemMessage))
+                return ((ChatRequestSystemMessage)message).Content;
+            if (message.GetType() == typeof(ChatRequestToolMessage))
+                return ((ChatRequestToolMessage)message).Content;
+            if (message.GetType() == typeof(ChatRequestUserMessage))
+                return ((ChatRequestUserMessage)message).Content;
+            throw new NotSupportedException("The message type is not supported");
+        }
+
+        /// <summary>
+        /// Get the activity, if the telemetry is enabled. If there is no listeners for
+        /// Activity source, null will be returned.
+        /// </summary>
+        /// <param name="name">The activity name.</param>
+        /// <returns>Activity or null.</returns>
+        private static Activity GetActivityMaybe(string name)
+        {
+            Activity act = null;
+            string enable = Environment.GetEnvironmentVariable(EnvironmentVariableSwitchName);
+            if (enable != null && (enable.Equals("true", StringComparison.OrdinalIgnoreCase) || enable.Equals("1")))
+                act = s_chatSource.CreateActivity(name, ActivityKind.Client);
+            return act;
+        }
+
+        /// <summary>
+        /// Set the tag on the activity, if the tag is present.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="name">The name of tag to set.</param>
+        /// <param name="value">Nullable value to be set.</param>
+        private void SetTagMaybe<T>(string name, Nullable<T> value) where T: struct
+        {
+            if (value.HasValue && m_activity != null)
+                m_activity.SetTag(name, value.Value);
+        }
+
+        /// <summary>
+        /// Return true if the activity for OpenTelemetry exists.
+        /// </summary>
+        public bool IsEnabled { get => m_activity != null; }
```

Comment:
is this used somewhere except tests? If not, please remove it and replace with tests that no telemetry is emitted (or that activity is null).

Example: 
https://github.com/openai/openai-dotnet/blob/f54b75226f6863426b0c728fc1635150028bcc2a/tests/Telemetry/ChatTelemetryTests.cs#L39

Also, scope is enabled if any of the signals are (activity or metrics)

Reply:
Fixed.